tests/IntegrationTests/WebAppUiTests/B2CWebAppCallsWebApiLocally.cs

```diff
@@ -0,0 +1,167 @@
+        private readonly Uri _keyvaultUri = new("https://webappsapistests.vault.azure.net");
+        private readonly ITestOutputHelper _output;
+        private readonly string _testAssemblyPath = typeof(B2CWebAppCallsWebApiLocally).Assembly.Location;
+
+        public B2CWebAppCallsWebApiLocally(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        [SupportedOSPlatform("windows")]
+        public async Task ChallengeUser_B2cFlow_LocalApp_ValidEmailPasswordCreds_TodoAppFunctionsCorrectly()
+        {
+            // Web app and api environmental variable setup.
+            string clientSecret = await UiTestHelpers.GetValueFromKeyvaultWitDefaultCreds(_keyvaultUri, KeyvaultClientSecretName);
+            var serviceEnvVars = new Dictionary<string, string>
+            {
+                {"ASPNETCORE_ENVIRONMENT", "Development" },
+                {TC.KestrelEndpointEnvVar, TC.HttpStarColon + TodoListServicePort}
+            };
+            var clientEnvVars = new Dictionary<string, string>
+            {
+                {"ASPNETCORE_ENVIRONMENT", "Development"},
+                {"AzureAdB2C__ClientSecret", clientSecret},
+                {TC.KestrelEndpointEnvVar, TC.HttpsStarColon + TodoListClientPort}
+            };
+
+            // Start the web app and api processes.
+            // Five second delay prevents transient issue where client fails to load on devbox the first time the test is run in VS after rebuilding.
+            Process? serviceProcess = UiTestHelpers.StartProcessLocally(_testAssemblyPath, _devAppPath + TC.s_todoListServicePath, TC.s_todoListServiceExe, serviceEnvVars);
+            Thread.Sleep(5000);
+            Process? clientProcess = UiTestHelpers.StartProcessLocally(_testAssemblyPath, _devAppPath + TC.s_todoListClientPath, TC.s_todoListClientExe, clientEnvVars);
+
+            // Get email and password from keyvault.
+            string email = await UiTestHelpers.GetValueFromKeyvaultWitDefaultCreds(_keyvaultUri, KeyvaultEmailName);
+            string password = await UiTestHelpers.GetValueFromKeyvaultWitDefaultCreds(_keyvaultUri, KeyvaultPasswordName);
+
+            // Playwright setup. To see browser UI, set 'Headless = false'.
+            const string TraceFileName = TraceClassName + "_TodoAppFunctionsCorrectly";
+            using IPlaywright playwright = await Playwright.CreateAsync();
+            IBrowser browser = await playwright.Chromium.LaunchAsync(new() { Headless = true });
+            IBrowserContext context = await browser.NewContextAsync(new BrowserNewContextOptions { IgnoreHTTPSErrors = true });
+            await context.Tracing.StartAsync(new() { Screenshots = true, Snapshots = true, Sources = true });
+
+            try
+            {
+                if (!UiTestHelpers.ProcessesAreAlive(new List<Process>() { clientProcess!, serviceProcess! }))
```

Comment:
this isn't quite right for nullables, the helper should either not return null, or they should be checked for null

Reply:
Can you help me understand why is it not correct for the helper to handle the null check?

Reply:
which helper?

Reply:
[ProcessIsAlive](https://github.com/AzureAD/microsoft-identity-web/blob/2091a0ac23a2b842d5263568af677ce89637a6fc/tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs#L271C15-L271C15)

Reply:
Synced offline and will change to move the null check to the StartProcessLocally helper method where the process is first initialized.